specs/protocol/holocene/derivation.md

```diff
@@ -0,0 +1,186 @@
+# Holocene Chain Derivation Changes
+
+# Stricter Derivation Rules
+
+The Holocene hardfork introduces several changes to batch derivation rules that render the
+derivation pipeline stricter and simpler.
+The changes are:
+
+- _Steady Batch Derivation_ derives invalid singluar batches immediately as deposit-only
+blocks.
+- _Partial Span Batch Validity_ determines the validity of singular batches within a span batch
+individually, only invalidating the remaining span batch upon the first invalid batch.
+- _Strict Batch Ordering_ required batches within and across channels to be strictly ordered.
+
+The channel and frame format are not changed, see also [A note on the current channel
+format](#a-note-on-the-current-channel-format) for a discussion about a possible future
+simplification of the channel and frame format under the stricter derivation rules.
+
+## Steady Batch Derivation
+
+Steady Batch Derivation changes the derivation rules for invalid batches, and instead of giving them
+a second chance for a worst-case full sequencing window, they are immediately derived as
+deposit-only blocks. There are two stages at which this new rule takes effect, in the batch queue
+and in the engine queue.
+
+### Batch Queue
+
+With Holocene, whereever a singular batch would be dropped before, it is now immediately replaced
+by a deposit-only batch.
+
+TODO: details
+
+### Engine Queue
+
+Holocene changes the way in which the Engine Queue responds to payload attributes errors: On a
+payload processing error, the payload is immediately replaced by the respecive empty deposit-only
```

Comment:
What sorts of processing errors can result in this?

Possible things?
- invalid block (tx with wrong nonce, tx with not enough balance for fees)
- invalid parent hash
- error responses here: https://github.com/ethereum/execution-apis/blob/main/src/engine/paris.md#response-1

Reply:
We do not list the types of errors that can cause this in the original spec either. There, we just say
> - On payload processing errors the attributes must be dropped, and the forkchoice state must be left unchanged.
>   - Eventually the derivation pipeline will produce alternative payload attributes, with or without batches.
>   - If the payload attributes only contained deposits, then it is a critical derivation error if these are invalid.

Do you think we should start listing the types of failures that can cause this?